server/user_test.go

```diff
@@ -61,3 +63,47 @@ func TestRouteUserStart(t *testing.T) {
 		})
 	}
 }
+
+func TestGetJiraUserFromMentions(t *testing.T) {
+	p := Plugin{}
+	p.userStore = getMockUserStoreKV()
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+
+	tests := map[string]struct {
+		mentions   *model.UserMentionMap
+		userSearch string
+	}{
+		"if no mentions, no users are returned": {
+			mentions: &model.UserMentionMap{}, userSearch: "join"},
+		"non connected user won't appear when mentioned": {
+			mentions: &model.UserMentionMap{
+				"non_connected_user": "non_connected_user",
+			}, userSearch: "non_connected_user"},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			user := p.GetJiraUserFromMentions(
+				testInstance1.InstanceID, *tc.mentions, tc.userSearch)
+			assert.Nil(t, user)
+		})
+	}
+}
+
+func TestGetJiraUserFromMentionsConnectReturnValidUser(t *testing.T) {
```

Comment:
The function being tested here is pretty small, and doesn't need to handle many cases. I suggest we collapse the two test functions here into one test function for readability. Instead of `assert.Nil` and `assert.NotNil`, you can assert that the value is equal to a given struct in test setup data. Then it should be easier to have all the tests in one test function.

Reply:
Do you see an opportunity to test the `executeAssign` function? This feature will potentially be used much more with this improvement, so having coverage on that function would be very helpful

Reply:
I fixed the first comment, I'll write the tests for executeAssign!